cmd/plugin/validate.go

```diff
@@ -46,13 +48,38 @@ func (v *validateCommand) PreRunE(_ *cobra.Command, _ []string) error {
 	return nil
 }
 
+func (v *validateCommand) overrideFile(plugin models.CommandLineMod, pluginPath string) error {
+	if v.pluginVersion == "" {
+		return nil
+	}
+	pluginRef := plugin.(*yaml.PluginSpec)
+	pluginRef.PluginVersion = v.pluginVersion
+	data, marshalErr := yml.Marshal(pluginRef)
+	if marshalErr != nil {
+		return marshalErr
+	}
+	ioerr := ioutil.WriteFile(pluginPath, data, 0)
+	if ioerr == nil {
+		v.logger.Info(fmt.Sprintf("Success in Overriding pluginversion in %s", pluginPath))
+	}
+	return ioerr
+}
+
 func (v *validateCommand) validateFile(pluginPath string) error {
-	v.logger.Info("validatig " + pluginPath)
+	v.logger.Info("\nValidating " + pluginPath)
 	if filepath.Ext(pluginPath) != ".yaml" {
 		return errors.New("expecting .yaml file at " + pluginPath)
 	}
-	plugin, err := yaml.NewPlugin(pluginPath)
-	v.logPluginAsYaml(&plugin)
+	var err error
+	var plugin models.CommandLineMod
+	if plugin, err = yaml.NewPluginSpec(pluginPath); err == nil {
```

Comment:
Reduce nesting, check `err != nil` and return err in that case. Outdent rest of the block.